### ISSUE TITLE
How to make Proxies on 4G SIM Mobile Network?

### ISSUE THREAD

**Issue by TheScarface** [z3APA3A/3proxy#343]
Hello! I am trying to use 3Proxy to make the Proxies based upon the Mobile 4G Connection.

Visual Explanation: https://i.imgur.com/CW0YVZb.png (Chart)

Problem: Mobile 4G Connection somehow uses Dual NAT (Double NAT) which doesn't allow port-forwarding.

Solution: I can use my Home Broadband for that. So basically, Home Broadband will act as the Proxy Address but it will use Mobile 4G Connection at the end. (Maybe it is known as Back-connect?)

Issue: I do not know how to set this up. Can anyone guide me?

Thanks

**Comment by z3APA3A** [comment]
https://3proxy.ru/howtoe.asp#CONNBACK

**Comment by TheScarface** [comment]
Yeah!, I was reading that only @z3APA3A. Thanks. But I have further questions.

1. How many systems do I need for this setup?

Thanks.

**Comment by z3APA3A** [comment]
You need host accessible from Internet to be used as connect back target.

**Comment by TheScarface** [comment]
Sorry, But I am rookie (new in all this). :( If you could say in little bit easier?

**Comment by z3APA3A** [comment]
If you have a real IP from provider, this IP is reachable from internet and you can forward an external port on you broadband connection or you can setup 3proxy on the broadband router directly, you can use it as a connect back target.

**Comment by TheScarface** [comment]
I want to do this:

Mobile IP --- > Home Broadband (Public IP) with Port ---- > VPS.

So, I will add the Home Broadband Public IP with Port on the VPS, but it will show the Mobile IP to the services such as WhatismyIP or YouTube.com.

I think people call it Reverse Proxy?

Also, Do you have Discord or something?

**Comment by TheScarface** [comment]
> https://3proxy.ru/howtoe.asp#CONNBACK

I am not clear with this part though: https://i.imgur.com/3XewxjG.png

So, How would you do?

**Comment by z3APA3A** [comment]
In order to use broadband connection, you need a device connected to broadband connection capable to run 3proxy.

**Comment by TheScarface** [comment]
> In order to use broadband connection, you need a device connected to broadband connection capable to run 3proxy.

Yeah, I have explained my setup here: 
![screenshot_917](https://user-images.githubusercontent.com/4048189/55416698-ed81cd00-558c-11e9-84c8-f9f75cc28491.png)

**Comment by z3APA3A** [comment]
If you have 2 connections at the same time, you do not need any connect back. You just need to use broadband connection as "internal" and mobile connection as "external" interface for proxy.

**Comment by TheScarface** [comment]
> If you have 2 connections at the same time, you do not need any connect back. You just need to use broadband connection as "internal" and mobile connection as "external" interface for proxy.

What is the direct link to the how-to for this specific operation? (I don't know what you call it in your technical term)

**Comment by z3APA3A** [comment]
See "internal" and "external" commands description in manual.
This may be helpful:
https://3proxy.ru/faqe.asp#INTEXT

**Comment by TheScarface** [comment]
> See "internal" and "external" commands description in manual.
> This may be helpful:
> https://3proxy.ru/faqe.asp#INTEXT

Do I need to do anything with the Route Tables or IP Tables of the Linux?

**Comment by z3APA3A** [comment]
It depends on current networking settings. If ip forwarding is enabled, you may be required to configure source routing or to bind 3proxy to interfaces (-Di / -De)

**Comment by TheScarface** [comment]
Hi, I did the internal and external interface thing as mentioned in the manual.

But, When I check the https://whoer.net/ for the Proxy. It shows the IP and the ISP of the Broadband and not of the 4G Mobile Dongle.

```
192.168.1.100 is Dongle 4G
192.168.3.3 is Home Broadband Modem
```

![screenshot_936](https://user-images.githubusercontent.com/4048189/55420267-3c7f3080-5594-11e9-85a1-de9b5c51d1d8.png)

**Comment by z3APA3A** [comment]
> If ip forwarding is enabled, you may be required to configure source routing or to bind 3proxy to interfaces (-Di / -De)

**Comment by xapanyun** [comment]
Actully it is very easy with app allproxy, we can get it from google play, then you will get your 4g proxy of your phone

**Comment by tannd88** [comment]
Hi all,
I need help configuring the 3proxy.cfg
![wlan0](https://user-images.githubusercontent.com/55658199/65387854-50eafb80-dd75-11e9-8685-2c9e3afecd29.jpg)
![image](https://user-images.githubusercontent.com/55658199/65387970-855eb780-dd75-11e9-8910-9ea710211004.png)
error service proxy after restart 3proxy
![image](https://user-images.githubusercontent.com/55658199/65388098-c48d0880-dd75-11e9-881a-9275ff05ef21.png)

Help me please!!!

**Comment by z3APA3A** [comment]
-De is implemented in devel (0.9) version
